tests/BookReader.test.js

```diff
@@ -1,59 +1,154 @@
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
 
-beforeEach(() => {
-  clock = sinon.useFakeTimers();
+  afterEach(() => {
+    clock.restore();
+  });
+
+  test("testing debounce", () => {
+    const func = jest.fn();
+    const debouncedFunc = BookReader.util.debounce(func, 1000);
+    // Call it immediately
+    debouncedFunc();
+    expect(func).toHaveBeenCalledTimes(0); // func not called
+
+    // Call it several times with 500ms between each call
+    for (let i = 0; i < 10; i++) {
+      clock.tick(500);
+      debouncedFunc();
+    }
+    expect(func).toHaveBeenCalledTimes(0); // func not called
+
+    // wait 1000ms
+    clock.tick(1000);
+    expect(func).toHaveBeenCalledTimes(1); // func called
+  });
 });
 
-afterEach(() => {
-  clock.restore();
+describe('Testing Throttle function', () => {
+  let clock;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
+  afterEach(() => {
+    clock.restore();
+  });
+
+  test('testing throttle', () => {
+    const callMethod = jest.fn();
+    const throttled = BookReader.util.throttle(() => callMethod(), 100);
+
+    const t = setInterval(() => {
+       throttled()
+    }, 50);
+
+    clock.tick(400);
+    expect(callMethod).toHaveBeenCalledTimes(4);
+    clearInterval(t);
+  });
 });
 
-test('testing debounce', () => {
-  const func = jest.fn();
-  const debouncedFunc = BookReader.util.debounce(func, 1000);
-  // Call it immediately
-  debouncedFunc();
-  expect(func).toHaveBeenCalledTimes(0); // func not called
+describe("Trigger function test", () => {
+  beforeEach(() => {
+    BookReader.prototype.trigger('SKR', 'val');
+  });
+  test("testing trigger", () => {
+    $(document).trigger('BookReader:' + 'SKR', 'val');
+  });
+});
 
-  // Call it several times with 500ms between each call
-  for(let i = 0; i < 10; i++) {
-    clock.tick(500);
-    debouncedFunc();
-  }
-  expect(func).toHaveBeenCalledTimes(0); // func not called
+describe("Testing extendParams", () => {
+  const obj1 = {
+    prop1: "SK",
+    prop2: 20
+  };
+  const obj2 = {
+    prop2: 19,
+    prop3: "SKR"
+  };
+  var modifiedNewParams = $.extend({}, obj2);
+  beforeEach(() => {
+    BookReader.prototype.extendParams(obj1, obj2);
+  });
+  test("jquery.extend", () => {
+    $.extend(obj1, modifiedNewParams);
+  });
+});
+
+describe("Testing jquery bind", () => {
+  function you() {
+    alert( "User clicked on 'you.'" );
+  };
+  beforeEach(() => {
+    BookReader.prototype.bind('SK', you);
+  });
+  test("jquery.bind", () => {
+    $(document).bind('BookReader:' + 'SK', you);
+  });
+});
 
-  // wait 1000ms
-  clock.tick(1000);
-  expect(func).toHaveBeenCalledTimes(1);  // func called
+describe("Testing unbind function", () => {
+  function fox() {
+    alert( "The quick brown fox jumps over the lazy dog." );
+  };
+  beforeEach(() => {
+    BookReader.prototype.unbind('SKR', fox);
+  });
+  test("jquery unbind", () => {
+    $(document).bind('BookReader:' + 'SKR', fox);
+  });
 });
```

Comment:
What are you testing in these `bind` and `unbind` tests? Are you testing to see whether the jQuery `bind` and `unbind` functionality works? If so, I wouldn't worry about that. That is functionality that is "internal" to jQuery so I don't think we need to make sure that it works.

Reply:
I added it to cover the coverage!